### PR TITLE
fix: type casting logic in ndarray validate function

### PR DESCRIPTION
**Pull request by agaraman0**
fixes issue #1439

### CODE REVIEW

File: docarray/typing/tensor/ndarray.py [hunk 1]
```diff
@@ -116,6 +117,8 @@ def validate(
             return cls._docarray_from_native(value)
         elif isinstance(value, NdArray):
             return cast(T, value)
+        elif isinstance(value, torch.Tensor):
+            return cast(T, value)
```
[the remark this answers is not in the document]
Comment (agaraman0) [reply docarray/typing/tensor/ndarray.py:121]: 1. I tried this change to test this for example [issue](https://github.com/docarray/docarray/issues/1439) change worked fine
2. I was trying to follow earlier `if comment `and wanted to make minimal change according to that

[the remark this answers is not in the document]
Comment (samsja) [reply docarray/typing/tensor/ndarray.py:121]: I see. then probably you should add one more line where you convert the torch tensor the a ndarray by calling to_numpy() on the torch tensor and calling NdArray._docarray_from_native()

[the remark this answers is not in the document]
Comment (agaraman0) [reply docarray/typing/tensor/ndarray.py:121]: got it, will fix it.

**Comment by samsja** [comment]
hey @agaraman0 seems that you have some import problem. When you import torch you need to be carefull

File: docarray/typing/tensor/ndarray.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 )
 
 import numpy as np
+import torch
```
Comment (samsja) [inline docarray/typing/tensor/ndarray.py:15]: you cannot do this directly

File: docarray/typing/tensor/ndarray.py (hunk 1 above)
Comment (samsja) [inline docarray/typing/tensor/ndarray.py:121]: you should rather import torch here.

be sure to use our utils `import_library`.

Comment (samsja) [reply docarray/typing/tensor/ndarray.py:121]: you can take inspiration here : https://github.com/docarray/docarray/blob/b3649b4b0f020887397f96e757711f71095b7fee/docarray/array/doc_vec/doc_vec.py#L122

**Pull request closed without merging**